stripe/src/test/java/com/stripe/android/util/StripeTextUtilsTest.java

```diff
@@ -245,4 +244,183 @@ public void asFundingType_whenUnknown_returnsUnknown() {
+        assertEquals("0000", groups[1]);
+        assertEquals("0232", groups[2]);
+        assertEquals("37", groups[3]);
+    }
+
+    @Test
+    public void separateCardNumberGroups_withAmexPrefix_returnsPrefixGroups() {
+        String testCardNumber = "378282246310005";
+        String[] groups = StripeTextUtils.separateCardNumberGroups(
+                        testCardNumber.substring(0, 2), Card.AMERICAN_EXPRESS);
+        assertEquals(3, groups.length);
+        assertEquals("37", groups[0]);
+        assertNull(groups[1]);
+        assertNull(groups[2]);
+
+        groups = StripeTextUtils.separateCardNumberGroups(
+                        testCardNumber.substring(0, 5), Card.AMERICAN_EXPRESS);
+        assertEquals(3, groups.length);
+        assertEquals("3782", groups[0]);
+        assertEquals("8", groups[1]);
+        assertNull(groups[2]);
+
+        groups = StripeTextUtils.separateCardNumberGroups(
+                testCardNumber.substring(0, 11), Card.AMERICAN_EXPRESS);
+        assertEquals(3, groups.length);
+        assertEquals("3782", groups[0]);
+        assertEquals("822463", groups[1]);
+        assertEquals("1", groups[2]);
+    }
+
+    @Test
+    public void separateCardNumberGroups_withVisaPrefix_returnsCorrectGroups() {
+        String testCardNumber = "4000056655665556";
+        String[] groups = StripeTextUtils.separateCardNumberGroups(
+                testCardNumber.substring(0, 2), Card.VISA);
+        assertEquals(4, groups.length);
+        assertEquals("40", groups[0]);
+        assertNull(groups[1]);
+        assertNull(groups[2]);
+        assertNull(groups[3]);
+
+        groups = StripeTextUtils.separateCardNumberGroups(
+                testCardNumber.substring(0, 5), Card.VISA);
+        assertEquals(4, groups.length);
+        assertEquals("4000", groups[0]);
+        assertEquals("0", groups[1]);
+        assertNull(groups[2]);
+        assertNull(groups[3]);
+
+        groups = StripeTextUtils.separateCardNumberGroups(
+                testCardNumber.substring(0, 9), Card.VISA);
+        assertEquals(4, groups.length);
+        assertEquals("4000", groups[0]);
+        assertEquals("0566", groups[1]);
+        assertEquals("5", groups[2]);
+        assertNull(groups[3]);
+
+        groups = StripeTextUtils.separateCardNumberGroups(
+                testCardNumber.substring(0, 15), Card.VISA);
+        assertEquals(4, groups.length);
+        assertEquals("4000", groups[0]);
+        assertEquals("0566", groups[1]);
+        assertEquals("5566", groups[2]);
+        assertEquals("555", groups[3]);
+    }
+
+    @Test
+    public void convertToSpacelessNumber_withSpacesInInterior_returnsSpacelessNumber() {
+        String testCardNumber = "4242 4242 4242 4242";
+        assertEquals("4242424242424242", StripeTextUtils.convertToSpacelessNumber(testCardNumber));
+    }
+
+    @Test
+    public void convertToSpacelessNumber_withExcessiveSpacesInInterior_returnsSpacelessNumber() {
```

Comment:
Please add a test for spaces on either side --- `convertToSpacelessNumber("                         12345     ")`